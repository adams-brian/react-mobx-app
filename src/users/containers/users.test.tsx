import { mount, shallow } from 'enzyme';
import * as React from 'react';
import { MemoryRouter } from 'react-router';
import * as sinon from 'sinon';

import UsersState from '../state';
import Users from './users';

let usersState: UsersState;

beforeEach(() => {
  usersState = new UsersState({
    createUser: sinon.fake(),
    deleteUser: sinon.fake(),
    loadUsers: sinon.fake.resolves([]),
    updateUser: sinon.fake()
  });
  sinon.spy(usersState, 'loadUsers');
});

describe("users", () => {
  it('renders as expected', () => {
    const component = shallow(<Users usersState={usersState} />);
    expect(component).toMatchSnapshot();
  });
  it('calls loadUsers', () => {
    expect((usersState.loadUsers as sinon.SinonSpy).called).toBe(false);
    mount(
      <MemoryRouter>
        <Users usersState={usersState} />
      </MemoryRouter>
    );
    expect((usersState.loadUsers as sinon.SinonSpy).calledOnce).toBe(true);
  });
});
