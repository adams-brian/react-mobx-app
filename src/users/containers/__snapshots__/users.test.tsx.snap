// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users renders as expected 1`] = `
<loading(Users)
  usersState={
    UsersState {
      "api": Object {
        "createUser": [Function],
        "deleteUser": [Function],
        "loadUsers": [Function],
        "updateUser": [Function],
      },
      "users": Array [],
    }
  }
/>
`;
